App_Use_Cases.tex

```diff
@@ -0,0 +1,447 @@
+
+
+Notifying an event:
+\pmixCodeImportC[]{sources/_autogen_/hybrid-prog-model.c_notify_event}
+
+\littleheader{Related Interfaces and Constants}
+
+{\large \refapi{PMIx_Notify_event}}
+\pasteSignature{PMIx_Notify_event}
+
+\refconst{PMIX_MODEL_DECLARED} \\
+\refconst{PMIX_MODEL_RESOURCES} \\
+\refconst{PMIX_OPENMP_PARALLEL_ENTERED} \\
+\refconst{PMIX_OPENMP_PARALLEL_EXITED} \\
+
+{\large \refapi{PMIx_Register_event_handler}}
+\pasteSignature{PMIx_Register_event_handler}
+
+{\large \refapi{pmix_event_notification_cbfunc_fn_t}}
+\pasteSignature{pmix_event_notification_cbfunc_fn_t}
+
+\refconst{PMIX_EVENT_ACTION_COMPLETE}
+
+
+
+\subsubsection{Coordinating at Runtime with Multiple Event Handlers}
+
+Coordinating with a threading library such as OpenMP creates the need for separate event handlers for threads of the same process.  For example in an MPI+OpenMP hybrid application, the MPI thread and the main OpenMP thread may both want to be notified anytime an OpenMP worker thread enters a parallel region.  This requiring support for multiple threads to potentially register different event handlers against the same status code.
+
+Multiple event handlers registered against the same event are processed in a chain-like manner based on the order in which they were registered, as modified by directive. Registrations against specific event codes are processed first, followed by registrations against multiple event codes and then any default registrations. At each point in the chain, an event handler is called by the PMIx progress thread and given a function to call when that handler has completed its operation. The handler callback notifies PMIx that the handler is done, returning a status code to indicate the result of its work. The results are appended to the array of prior results, with the returned values combined into an array within a single pmix_info_t as follows:
+\begin{itemize}
+\item \texttt{array[0]}: the event handler name provided at registration (may be an empty field if a string name was not given) will be in the key, with the pmix_status_t value returned by the handler
+\item \texttt{array[*]}: the array of results returned by the handler, if any.
+\end{itemize}
+
+The current PMIx standard does not actually specify a default ordering for event handlers as they are being registered. However, it does include an inherent ordering for invocation. Specifically, PMIx stipulates that handlers be called in the following categorical order:
+
+\begin{itemize}
+\item single status event handlers - i.e., handlers that were registered against a single specific status.
+\item multi status event handlers - those registered against more than one specific status
+\item default event handlers - those registered against no specific status
+\end{itemize}
+
+\littleheader{Code Example}
+
+From the OpenMP master thread:
+
+\pmixCodeImportC[]{sources/_autogen_/hybrid-prog-model.c_omp_thread}
+
+From the MPI thread:
```

Comment:
What is meant with the MPI thread? MPI only knows about processes, unless the main thread is meant, which refers to the thread that called MPI_Init_thread
```suggestion
From the MPI process:
```